aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -104,16 +115,23 @@ pub(crate) fn run_script_prologue(
     traversal_context: &mut TraversalContext,
     is_simulation: bool,
 ) -> Result<(), VMStatus> {
-    let txn_sequence_number = txn_data
-        .replay_protector()
-        .get_sequence_number()
-        .unwrap_or(0);
+    let txn_replay_protector = txn_data.replay_protector();
     let txn_authentication_key = txn_data.authentication_proof().optional_auth_key();
     let txn_gas_price = txn_data.gas_unit_price();
     let txn_max_gas_units = txn_data.max_gas_amount();
     let txn_expiration_timestamp_secs = txn_data.expiration_timestamp_secs();
     let chain_id = txn_data.chain_id();
     let mut gas_meter = UnmeteredGasMeter;
+
+    if !(features.is_transaction_payload_v2_enabled() && features.is_orderless_txns_enabled()) {
+        if let ReplayProtector::Nonce(_) = txn_replay_protector {
+            return Err(VMStatus::error(
+                StatusCode::FEATURE_NOT_ENABLED,
+                Some("Orderless transactions is not yet supported".to_string()),
+            ));
+        }
+    }
```

Comment:
This should be done in validation phase.

Reply:
I have a similar assertion in aptos_vm validation method as well. Is this what you mean?

https://github.com/aptos-labs/aptos-core/pull/16299/files#diff-fd22752e67147bf54afdcfa60f8074d0d6558f1610b34a940e6591d7bbdaec45R1678-R1698

Reply:
Yeah, the transactions are discarded in aptos_vm before hitting this. May not make much difference. I just thought, we don't prefer assert! statements in the main execution code. So, I returned an error instead of asserting.